students_proj/src/main/java/com/musala/simple/students/Main.java

```diff
@@ -0,0 +1,279 @@
+
+		if (inputIsValid(args)) {
+
+			String studentsJsonInfo = readFile(args[0]);
+
+			if (!isValidJson(studentsJsonInfo)) {
+				System.err.println(ERROR_NOT_VALID_JSON);
+				try {
+					readBackupJson();
+				} catch (ResourceDoesNotExistException e) {
+					e.printStackTrace();
+				}
+			} else {
+				StudentWrapper studentWrapper = new Gson().fromJson(studentsJsonInfo, StudentWrapper.class);
+				StudentGroup studentGroup = new StudentGroup();
+				studentGroup.fillStudentGroup(studentWrapper.students);
+
+				Map<Integer, Student> studentsMap = studentGroup.getStudents();
+
+				int requestedStudentId = readStudentId(args);
+				boolean studentExists = false;
+				if (requestedStudentId != -1) {
+					try {
+						Student student = studentGroup.getStudentById(requestedStudentId);
+						studentExists = true;
+						printStudentDetails(student);
+					} catch (IllegalArgumentException iae) {
+						System.err.println(ERROR_STUDENT_NOT_FOUND);
+					}
+				}
+				if (!studentExists) {
+					List<Student> studentsList = studentsMap.values().stream().collect(Collectors.toList()); 
+					printStudents(studentsList);
+				}
+
+				updateBackupFile(studentsJsonInfo);
+			}
+		} else {
+			try {
+				readBackupJson();
+			} catch (ResourceDoesNotExistException e) {
+				e.printStackTrace();
+			}
+		}
+	}
+
+	/**
+	 * Validates the command line input arguments.
+	 *
+	 * @param args  A String array with command line arguments
+	 * @return true if the first argument in the array
+	 * is a valid path to a file with extension ".json".
+	 */
+	private static boolean inputIsValid(String[] args) {
+
+		/* Check if any arguments are provided */
+		if (args.length == 0) {
+			System.err.println(ERROR_PATH_MISSING);
+			return false;
+		}
+
+		String argPath = args[0];
+		File file = new File(argPath);
+
+		/* Check if path exists */
+		if (!file.exists()) {
+			System.err.println(ERROR_INVALID_PATH);
+
+			return false;
+		}
+
+		/* Check if the valid path is a file */
+		if (!file.isFile()) {
+			System.err.println(ERROR_NOT_A_FILE);
+
+			return false;
+		}
+
+		/* Check the file extension to be JSON. */
+		if (!argPath.endsWith(JSON_FILE_EXTENSION)) {
+			System.err.println(ERROR_FILE_NOT_JSON);
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Creates a backup Json file using the students objects
+	 * provided as a Json String parameter. If a backup file
+	 * already exists, it's being overwritten.
+	 *
+	 * @param studentsJsonInfo A Json String, containing the students
+	 * to be written to the backup file.
+	 */
+	private static void createBackupJson(String studentsJsonInfo) {
+		try (FileWriter fileWriter = new FileWriter(BACKUP_JSON_PATH, false)) {
+			fileWriter.write(studentsJsonInfo);
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+	}
+
+	/**
+	 * Reads the backup Json file (if exists) and prints the
+	 * students information from it.
+	 *
+	 */
+	private static void readBackupJson() throws ResourceDoesNotExistException {
```

Comment:
Either name both methods `readBackupJson()` and `updateBackupJson()` or `readBackupFile()` and `updateBackupFile()` for consistency. :)

Reply:
I thought of that and I thought I did it for all three - create, read, update. Obviously I didn't :)

Reply:
So, stick to one version of your choice and apply it for all three methods. :)